### PR TITLE
Exception enum mapper

### PR DESCRIPTION
**Pull request by joey11111000111**
https://trello.com/c/2GRg0PDT

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -203,7 +203,7 @@
                         <phase>post-integration-test</phase>
                         <configuration>
                             <rules>
-                                <rule>
+                                <exceptionMessagePair>
```
Comment (lamfalusy) [inline pom.xml]: Be careful, you renamed a tag in pom.xml with your refactor, please revert it.

**Comment by lamfalusy** [comment]
Be careful, you modified web/src/main/webapp/resources/thirdparty/bootstrap/css/bootstrap.css.map file with your renaming refactor commit. Please revert it.

File: web/src/test/java/com/epam/rft/atsy/web/mapper/ExceptionMessagePairValidationExceptionMapperImplTest.java [hunk 2]
```diff
@@ -10,7 +10,7 @@
 import java.util.HashSet;
 import java.util.Set;
 
-public class RuleValidationExceptionMapperImplTest {
+public class ExceptionMessagePairValidationExceptionMapperImplTest {
```
Comment (lamfalusy) [inline web/src/test/java/com/epam/rft/atsy/web/mapper/ExceptionMessagePairValidationExceptionMapperImplTest.java]: Why did you rename this class? In this test class you test RuleValidationExceptionMapperImpl class, as a convention the test class name should be RuleValidationExceptionMapperImplTest.

File: web/src/main/java/com/epam/rft/atsy/web/configuration/ExceptionMapperConfiguration.java [hunk 3]
```diff
@@ -12,47 +12,68 @@
...
+/**
+ * This configuration class is responsible for the bean that stores all the validation exceptions
+ * along with their corresponding messages. The bean is filled with data here, so if some changes
+ * happen among the validation exceptions or the exception messages, this class must be adjusted
+ * according to the changes.
+ */
 @Configuration
 public class ExceptionMapperConfiguration {
 
+  /**
```
Comment (lamfalusy) [inline web/src/main/java/com/epam/rft/atsy/web/configuration/ExceptionMapperConfiguration.java:31]: Follow Javadoc conventions: write the first sentence as a short summary of the method, as Javadoc automatically places it in the method summary table (and index).

File: web/src/main/java/com/epam/rft/atsy/web/mapper/RuleValidationExceptionMapperImpl.java [hunk 4]
```diff
@@ -1,25 +1,37 @@
 package com.epam.rft.atsy.web.mapper;
 
+import com.epam.rft.atsy.web.configuration.ExceptionMapperConfiguration;
 import org.springframework.stereotype.Component;
 import org.springframework.util.Assert;
 
 import java.util.Set;
 import javax.annotation.Resource;
 
+/**
```
Comment (lamfalusy) [inline web/src/main/java/com/epam/rft/atsy/web/mapper/RuleValidationExceptionMapperImpl.java:10]: Follow Javadoc conventions: write the first sentence as a short summary of the method, as Javadoc automatically places it in the method summary table (and index).

File: web/src/main/java/com/epam/rft/atsy/web/configuration/ExceptionMapperConfiguration.java [hunk 5]
```diff
@@ -12,47 +12,68 @@
...
-import com.epam.rft.atsy.web.mapper.Rule;
+import com.epam.rft.atsy.web.mapper.ExceptionMessagePair;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
 import java.util.Collections;
 import java.util.HashSet;
 import java.util.Set;
 
+/**
```
Comment (lamfalusy) [inline web/src/main/java/com/epam/rft/atsy/web/configuration/ExceptionMapperConfiguration.java:23]: Follow Javadoc conventions: write the first sentence as a short summary of the class, as Javadoc automatically places it in the class summary table (and index).

File: web/src/main/java/com/epam/rft/atsy/web/mapper/RuleValidationExceptionMapper.java [hunk 6]
```diff
@@ -1,8 +1,20 @@
 package com.epam.rft.atsy.web.mapper;
 
+/**
```
Comment (lamfalusy) [inline web/src/main/java/com/epam/rft/atsy/web/mapper/RuleValidationExceptionMapper.java:3]: Follow Javadoc conventions: write the first sentence as a short summary of the calss, as Javadoc automatically places it in the class summary table (and index).

File: web/src/test/java/com/epam/rft/atsy/web/mapper/RuleValidationExceptionMapperImplTest.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+  @Test
+  public void getCorrectMessageKeyWhenExceptionIsValid() {
+    String
+        actualMessageKey =
+        ruleValidationExceptionMapper.getMessageKeyByException(VALID_EXCEPTION);
+    assertEquals(EXPECTED_MESSAGE_KEY, actualMessageKey);
+  }
+
+}//class
```
Comment (lamfalusy) [inline web/src/test/java/com/epam/rft/atsy/web/mapper/RuleValidationExceptionMapperImplTest.java]: Delete this "//class" comment, it is not a commenting convection what we follow.

File: web/src/main/java/com/epam/rft/atsy/web/configuration/ExceptionMapperConfiguration.java (hunk 3 above)
Comment (joey11111000111) [reply web/src/main/java/com/epam/rft/atsy/web/configuration/ExceptionMapperConfiguration.java:31]: The javadocs are without line breaks after the summary sentence because of the current code formatting conventions that we applied to this project.

File: web/src/main/java/com/epam/rft/atsy/web/mapper/RuleValidationExceptionMapperImpl.java (hunk 4 above)
Comment (joey11111000111) [reply web/src/main/java/com/epam/rft/atsy/web/mapper/RuleValidationExceptionMapperImpl.java:10]: The javadocs are without line breaks after the summary sentence because of the current code formatting conventions that we applied to this project.

File: web/src/main/java/com/epam/rft/atsy/web/configuration/ExceptionMapperConfiguration.java (hunk 5 above)
Comment (joey11111000111) [reply web/src/main/java/com/epam/rft/atsy/web/configuration/ExceptionMapperConfiguration.java:23]: The javadocs are without line breaks after the summary sentence because of the current code formatting conventions that we applied to this project.

File: web/src/main/java/com/epam/rft/atsy/web/mapper/RuleValidationExceptionMapper.java (hunk 6 above)
Comment (joey11111000111) [reply web/src/main/java/com/epam/rft/atsy/web/mapper/RuleValidationExceptionMapper.java:3]: The javadocs are without line breaks after the summary sentence because of the current code formatting conventions that we applied to this project.
